### PR TITLE
Language Picker + v2 launch resume

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**

### CODE REVIEW

File: src/components/Dropdown.vue [hunk 1]
```diff
@@ -56,6 +60,7 @@ export default {
       value: "",
     };
   },
+  created() {},
```
Comment (suryabulusu) [inline src/components/Dropdown.vue]: whats this for

File: src/components/NoClassMessage.vue [hunk 2]
```diff
@@ -24,5 +26,19 @@ export default {
       AFLogoSvg: assets.AFLogoSvg,
     };
   },
+  computed: {
+    displayMessage() {
+      return this.isLanguageSelectedEnglish
+        ? "There is no class right now."
```
Comment (suryabulusu) [inline src/components/NoClassMessage.vue:32]: we wanted to change this to `There is no session right now` na

File: src/pages/InformationForm.vue [hunk 3]
```diff
@@ -65,21 +68,25 @@ export default {
     },
   },
   async created() {
+    console.log(this.$store.state.sessionData);
```
Comment (suryabulusu) [inline src/pages/InformationForm.vue]: remove when merging

File: src/pages/LandingPage.vue [hunk 4]
```diff
@@ -1,11 +1,28 @@
...
+      </p>
+      <p v-if="isLanguageSelectedHindi" class="text-4xl">
+        पोर्टल में आपका स्वागत है!
+      </p>
+    </div>
   </div>
 </template>
 
 <script>
+import { mapState } from "vuex";
```
Comment (suryabulusu) [inline src/pages/LandingPage.vue]: where are you using this

File: src/pages/NewSignup.vue [hunk 5]
```diff
@@ -100,17 +104,27 @@ export default {
     Object.keys(this.formData.attributes).forEach((field) => {
       this.formData.attributes[field]["component"] =
         typeToInputParameters[this.formData.attributes[field].type];
+      this.formData.attributes[field]["show"] =
+        this.formData.attributes[field].showBasedOn == "" ? true : false;
+      this.formData.attributes[field]["required"] =
+        this.formData.attributes[field].required == "TRUE" ? true : false;
     });
+    console.log(typeof this.$store.state.sessionData.id_generation);
```
Comment (suryabulusu) [inline src/pages/NewSignup.vue]: remove when merging

File: src/pages/NewSignup.vue [hunk 6]
```diff
@@ -144,22 +160,46 @@ export default {
...
-      if (field.dependant) {
-        if (this.userData[field.dependantField])
-          return field.dependantFieldMapping[
-            this.userData[field.dependantField]
-          ];
-      } else return field.options;
+    showBasedOn() {
+      return Object.keys(this.formData.attributes).forEach((field) => {
+        let fieldAttributes = this.formData.attributes[field];
+        let showBasedOn = fieldAttributes.showBasedOn;
```
Comment (suryabulusu) [inline src/pages/NewSignup.vue:164]: is it fine to have method and variable with same name

**Review by suryabulusu — COMMENTED** [review]
left comments

File: src/pages/LandingPage.vue (hunk 4 above)
Comment (sudheshna-donthineni) [reply src/pages/LandingPage.vue]: removed

File: src/pages/NewSignup.vue (hunk 6 above)
Comment (sudheshna-donthineni) [reply src/pages/NewSignup.vue:164]: it's a local variable. Scope is not beyond the function. I don't see a problem.

File: src/components/NoClassMessage.vue (hunk 2 above)
Comment (sudheshna-donthineni) [reply src/components/NoClassMessage.vue:32]: Separate PR

File: src/pages/Home.vue [hunk 7]
```diff
@@ -11,14 +11,26 @@
...
     <NoClassMessage />
   </div>
-  <div v-if="isLandingPage">
+
+  <div v-if="isLandingPage" class="flex h-screen flex-col">
     <LandingPage />
   </div>
+
   <div v-else>
+    <LanguagePicker />
```
Comment (suryabulusu) [inline src/pages/Home.vue]: so language picker will be shown in both old and new flows?

File: src/pages/NewSignin.vue [hunk 8]
```diff
@@ -276,19 +309,12 @@ export default {
...
-            this.$store.state.sessionData.purpose["sub-type"],
-            this.$store.state.sessionData.platform,
-            this.$store.state.sessionData.platform_id,
             this.userInformation["student_id"],
-            this.getAuthTypes.toString(),
-            this.$store.state.groupData.name,
-            this.$store.state.groupData.input_schema.userType,
-            this.$store.state.sessionData.session_id,
-            "",
             "",
```
Comment (suryabulusu) [inline src/pages/NewSignin.vue:322]: what are these empty strings for

**Pull request merged**